frontend/src/metabase-lib/query.ts

```diff
@@ -82,6 +82,15 @@ export function replaceClause(
   return ML.replace_clause(query, stageIndex, targetClause, newClause);
 }
 
+export function swapClauses(
+  query: Query,
+  stageIndex: number,
+  sourceClause: Clause,
+  targetClause: Clause,
+): Query {
```

Comment:
Seeing that every `*Step` component has its own `handleReorder*` implementation, I thought that the benefit is type safety - it should not be possible to swap an aggregation clause with a breakout clause, etc.

We could enforce it here:

```suggestion
export function swapClauses<
  C extends
    | AggregationClause
    | ExpressionClause
    | BreakoutClause
    | FilterClause
    | OrderByClause,
>(query: Query, stageIndex: number, sourceClause: C, targetClause: C): Query {
```